### PR TITLE
Add pre-commit and the GitHub Super Linter

### PR DESCRIPTION
**Pull request by jbampton**
Official -> "Git hook scripts are useful for identifying simple issues before submission to code review. We run our hooks on every commit to automatically point out issues in code such as missing semicolons, trailing whitespace, and debug statements. By pointing these issues out before code review, this allows a code reviewer to focus on the architecture of a change while not wasting time with trivial style nitpicks." 

https://pre-commit.com/

So pre-commit for local tests and GitHub Super Linter for backup tests.  We can add more tests later.

Shows the final `pre-commit` run:

Screen Shot 2021-04-12 at 6 10 11 am

### CODE REVIEW

**Comment by rberrelleza** [comment]
This looks great, thanks @jbampton! Do you think we should update the developer guide with instructions on how to install pre-commit, and how to use it?

**Comment by rberrelleza** [comment]
> This looks great, thanks @jbampton! Do you think we should update the developer guide with instructions on how to install pre-commit, and how to use it?

Didn't realize that was already there. This is pretty cool, let me know when it's ready to be merged?

File: .github/linters/.markdown-lint.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+MD007: false
```
Comment (rberrelleza) [inline .github/linters/.markdown-lint.yml:1]: can we add comments on what each rule is?

**Review by rberrelleza — APPROVED** [review]

File: .github/linters/.markdown-lint.yml (hunk 1 above)
Comment (jbampton) [reply .github/linters/.markdown-lint.yml:5]: I added three comments at the top of this markdownlint config file.

I think it's best if we just link to the offical rules.

File: .github/linters/.yaml-lint.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+
+# https://yamllint.readthedocs.io/en/stable/index.html
```
Comment (jbampton) [inline .github/linters/.yaml-lint.yml:3]: I added two comments here for extra info too 👍

File: contributing.md [hunk 3]
```diff
@@ -130,4 +133,39 @@ make lint
...
+
+To run a single hook use `pre-commit run --all-files <hook_id>`
+
+To update use `pre-commit autoupdate`
+
+- [Quick start](https://pre-commit.com/#quick-start)
+- [Usage](https://pre-commit.com/#usage)
+- [pre-commit-autoupdate](https://pre-commit.com/#pre-commit-autoupdate)
+
+## Other Linters
```
Comment (jbampton) [inline contributing.md:160]: I added this new section too for other linters 🥇

File: .github/linters/.markdown-lint.yml (hunk 1 above)
Comment (jbampton) [reply .github/linters/.markdown-lint.yml:5]: Hey @rberrelleza also when you run `markdownlint` on the command line it prints out the line numbers with the failing issues.

This is an image from another repo:

Screen Shot 2021-04-13 at 8 12 20 pm

**Comment by rberrelleza** [comment]
Looks great, I'll fix the linting issues around bash in a separate script. Is this ready to merge @jbampton ?

**Comment by jbampton** [comment]
Ready to go @rberrelleza 👍

**Pull request merged**